client/src/components/ClientProfile/Information/Information.js

```diff
@@ -16,96 +13,190 @@ import VisibilityIcon from '@material-ui/icons/Visibility';
+  const PasswordIcon = passwordHidden ? VisibilityIcon : VisibilityOffIcon;
+  const [state, setState] = useState({
+    ...customerEmpty,
   });
 
   const preventDefault = (event) => event.preventDefault();
   const handleChange = (panel) => (event, isExpanded) => {
     setExpanded(isExpanded ? panel : false);
   };
-  // TODO: complete validation
-
   const onChange = (stateName) => (event) => {
     if (stateName === 'checkedSubscribe') {
-      setValues({ ...values, [stateName]: event.target.checked });
+      setState({ ...state, [stateName]: event.target.checked });
     } else {
-      setValues({ ...values, [stateName]: event.target.value });
+      setState({ ...state, [stateName]: event.target.value.trim() });
     }
   };
-
   const handleDateChange = (date) => {
-    setValues({ ...values, dateOfBirth: date });
+    setState({ ...state, birthdate: date });
+  };
+  // Modal window
+  const successSwalSaveContactInfo = (event) => {
+    Swal.fire(
+      'Saved!',
+      'Your changes have been saved.',
+      'success',
+    );
   };
 
-  const onSubmit = (event) => {
-    event.preventDefault();
+  const errorSwalSaveContactInfo = (message) => {
+    Swal.fire({
+      title: 'Error...',
+      text: message,
+    });
+  };
 
+  const editCustomerInfo = () => {
+    const { getCustomerInfo } = props;
     const {
-      email, password, firstName,
-      lastName, phone, address, birthdate,
+      firstName, lastName, telephone,
+      email, oldPassword, newPassword, address, birthdate,
       gender, haveChildren, haveCar, eBicycle,
       checkedSubscribe,
-    } = values;
-    // TODO: complete validation
-    // console.log('%c⧭ values', 'color: #00ff73', values);
-  };
+    } = state;
+    const updatedCustomer = {
+      firstName,
+      lastName,
+      telephone,
+      email,
+      address,
+      birthdate,
+      gender,
+      haveChildren,
+      haveCar,
+      eBicycle,
+      checkedSubscribe,
+    };
 
-  const handleClickShowPassword = () => {
-    setValues({ ...values, showPassword: !values.showPassword });
-  };
-  const handleMouseDownPassword = (event) => {
-    event.preventDefault();
-  };
+    const passwords = {
+      password: oldPassword,
+      newPassword,
+    };
 
-  // const updatedCustomer = {
-  //   firstName: "Antonio",
-  //   lastName: "Front End Developer"
-  // };
+    Object.keys(passwords).forEach((key) => (passwords[key] == null || passwords[key] === '') && delete passwords[key]);
+    if (passwords.newPassword && passwords.password) {
+      axios
+        .put(UPDATE_PASSWORD, passwords)
+        // eslint-disable-next-line no-shadow
+        .then((updatedCustomer) => {
+          successSwalSaveContactInfo();
+          getCustomerInfo();
+        })
+        .catch((err) => {
+          const { data } = err.response;
+          const message = JSON.stringify(data).replace(/{"/g, ' ').replace(/}/g, ' ').replace(/":/g, ':')
+            .replace(/,"/g, '; ');
```

Comment:
Strange part, why you just not get to keys from error object and not concatenate their values?

Reply:
It was done in an order to show what's the error with field.